client/wfnews-war/src/main/angular/src/app/services/common-utility.service.ts

```diff
@@ -164,16 +164,25 @@ valueMatch = trimmedAddress.substring(0, valueLength);
   }
 
   checkLocationServiceStatus(): Promise<boolean> {
-    const timeoutDuration = 4000; // 4 seconds limit
+    const timeoutDuration = 10000; // 10 seconds limit
 
     const timeoutPromise = new Promise<boolean>((resolve) => {
       setTimeout(() => resolve(false), timeoutDuration);
     });
-
-    const locationPromise = Geolocation.getCurrentPosition()
+    
+    let locationPromise = Geolocation.getCurrentPosition()
       .then(() => Promise.resolve(true))
       .catch(() => Promise.resolve(false));
 
+    // resolve for firefox
+    if (window?.navigator?.userAgent?.search("Firefox") && window?.navigator?.userAgent?.search("Firefox") > -1
```

Comment:
minor, maybe we can simplify this to 

if (window?.navigator?.userAgent?.includes("Firefox") && window?.navigator?.geolocation)

Reply:
For sure, updated it now. Thanks!